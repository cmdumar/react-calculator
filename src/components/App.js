import React from 'react';
import {
  BrowserRouter, Link, Route, Switch,
} from 'react-router-dom';
import About from './About';
import Calculator from './Calculator';
import Home from './Home';

function App() {
  return (
    <BrowserRouter>
      <nav>
        <ul className="navbar">
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/calculator">Calculator</Link>
          </li>
          <li>
            <Link to="/about">About</Link>
          </li>
        </ul>
      </nav>
      <Switch>
        <Route path="/calculator" component={Calculator} />
        <Route path="/about" component={About} />
        <Route path="/" component={Home} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
