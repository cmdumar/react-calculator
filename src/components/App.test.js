import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('App Component', () => {
  const app = renderer.create(<App />).toJSON();

  expect(app).toMatchSnapshot();
});
