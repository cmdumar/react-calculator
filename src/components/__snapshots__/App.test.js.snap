// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component 1`] = `
Array [
  <nav>
    <ul
      className="navbar"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="homepage"
  >
    <h1>
      Math Magician
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>,
]
`;
